### ISSUE TITLE
Does `findRandom` return an array of sequential documents?

### ISSUE THREAD

**Issue by joeytwiddle** [larryprice/mongoose-simple-random#8]
Would I be right in thinking that the "find many" function selects the first document randomly but the rest are simply those documents that directly follow it in mongo's sequence?

If so, it might be good to document that clearly. For some applications it would be an undesirable limitation. (For example pulling 5 random playing cards from an unshuffled deck would nearly always return a straight run!)

**Comment by larryprice** [comment]
I think that's how it currently works, and I agree that it's probably non-optimal as far as functionality goes. I'll happily look at any submitted PR.

**Comment by joeytwiddle** [comment]
No solution here I'm afraid. The approach I landed on was to choose one random document *n* times, using your `skip` approach, and then handle the risk of the same document being chosen twice. That's *n* queries!

From the research I did: an option some people used was to include a random value in a field of *every document* in the DB, and then use that field for ordering. But it seems to me that those values should be updated each time (at least for the retrieved documents), otherwise later queries may see similar repeated blocks of documents.

I wonder if a mongo aggregate query could somehow hash (the objid of each document along with a fresh random value supplied each time). In theory that could provide a random an unpredictable value we could sort on. It might be pretty heavy on large collections though.
